.github/workflows/release.yaml

```diff
@@ -203,21 +204,24 @@ jobs:
         if: github.event_name == 'workflow_dispatch'
         uses: docker/build-push-action@v6
         with:
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
+          cache-from: |
+            type=registry,ref=ghcr.io/metalbear-co/mirrord-cli:buildcache
+          cache-to: |
+            type=registry,ref=ghcr.io/metalbear-co/mirrord-cli:buildcache,mode=max
           context: .
           file: mirrord/cli/Dockerfile
           platforms: linux/amd64,linux/arm64
           push: true
           tags: |
             ghcr.io/metalbear-co/mirrord-cli-staging:${{ github.sha }}
-
       - name: Build and push (final/release)
         if: github.event_name != 'workflow_dispatch'
         uses: docker/build-push-action@v6
         with:
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
+          cache-from: |
+            type=registry,ref=ghcr.io/metalbear-co/mirrord-cli:buildcache
+          cache-to: |
+            type=registry,ref=ghcr.io/metalbear-co/mirrord-cli:buildcache,mode=max
```

Comment:
maybe this would be faster if the ci workflow would also use the same cache registry, and the branch would be used as a tag with a fallback to main (`/mirrord-cli:main` in a second cache-from line)?

That way on workflow_dispatch there is probably already a pretty good cache in the branch tag, and on final/release we can just directly use `main` as a tag, which would probably have very relevant cache already.

Reply:
Could be optimized in a separate PR though